app/controllers/newsletter_subscriptions_controller.rb

```diff
@@ -18,6 +29,14 @@ def create
   private
 
   def subscription_params
-    params.require(:subscription).permit(:email)
+    params.require(:subscription).permit(:email, :ga_categoryid, :category)
+  end
+
+  def categories
+    ['footer', 'sticky', 'in_article']
+  end
+
+  def set_cookie
```

Comment:
@tomas-stefano @nmiddleweek all the necessary changes how now happened so this is good to go